README.adoc

```diff
@@ -70,16 +82,314 @@ $ bundle exec liquidoc -d _data/data-sample.yml -t _templates/liquid/tpl-sample.
+link:https://help.shopify.com/themes/liquid/basics[*Liquid*] is used for parsing complex variable data, typically for iterated output.
+For instance, an data structure of glossary terms and definitions that needs to be looped over and pressed into a more publish-ready markup, such as Markdown, AsciiDoc, reStructuredText, LaTeX, or HTML.
+
+Any valid Liquid-formatted template is accepted, in the form of a text file with any extension.
+For data sourced in CSV format or extracted through regex source parsing, all data is passed to the Liquid template parser as an array called *data*, containing one or more rows to be iterated through.
+Data sourced in YAML, XML, or JSON may be passed as complex structures with custom names determined in the file contents.
+
+Looping through known data formats is fairly straightforward.
+A for loop iterates through your data, item by item.
+Each item or row contains one or more key-value pairs.
+
+[[rows_asciidoc]]
+.Example -- rows.asciidoc Liquid template
+[source,liquid]
+----
+{% for row in data %}{{ row.name }}::
+{{ row.description }}
++
+[horizontal.simple]
+Required:: {% if row.required == "true" %}*Yes*{% else %}No{% endif %}
+{% endfor %}
+----
+
+In <<rows_asciidoc>>, we're instructing Liquid to iterate through our data items, generating a data structure called `row` each time.
+The double-curly-bracketed tags convey variables to evaluate.
+This means `{{ row.name }}` is intended to express the value of the *name* parameter in the item presently being parsed.
+The other curious marks such as `::` and `[horizontal.simple]` are AsciiDoc markup -- they are the formatting we are trying to introduce to give the content form and semantic relevance.
+
+.Non-printing Markup
+****
+In Liquid and most templating systems, any row containing a non-printing “tag” will print leave a blank line in the output after parsing.
+For this reason, it is advised that you stack tags horizontally when you do not wish to generate a blank line, as with the first row above.
+A non-printing tag such as `{% endfor %}` will generate a blank line that is convenient in the output but likely to cause clutter here.
+
+This side effect of templating is unfortunate, as it discourages elegant, “accordian-style” code nesting, as in the HTML example below (<<parsed_html>>).
+In the end, ugly Liquid templates can generate elegant markup output exquisite precision.
+****
+
+The above would generate the following:
+
+[[asciidoc_formatted_source]]
+.Example -- AsciiDoc-formatted output
+[source,asciidoc]
+----
+A_B::
+A thing that *SnASFHE&"\|+1Dsaghf
++
+[horizontal.simple]
+Required::: *Yes*
+
+G_H::
+Some text for &hdf'" 1t`F
++
+[horizontal.simple]
+Required::: No
+----
+
+The generically styled AsciiDoc rich text reflects the distinctive structure with (very little) more elegance.
+
+.AsciiDoc rich text (parsed)
```

Comment:
nit: maybe rather than `parsed`, maybe say `rendered`?  Since github will render the asciidoc for us here

Reply:
That's actually exactly what I'm looking for. I've actually struggled over this, between "parsed", "processed", "generated", "compiled", and "rendered". I needed a second opinion, so your instinct is super informative.